include/ofi_mem.h

```diff
@@ -306,7 +306,7 @@ struct util_buf_footer {
 };
 
 union util_buf {
-	struct slist_entry entry;
+	struct dlist_entry entry;
```

Comment:
slist is more efficient than dlist.  Do we need to use dlist here?

Reply:
`slist` allows removing from head only, but we need to remove from different places.
Btw, I analyzed disassembled code
- `util_buf_alloc()` with slists:
```
000000000002b070 <util_buf_alloc>:
   2b070:	53                   	push   %rbx
   2b071:	48 8b 47 10          	mov    0x10(%rdi),%rax
   2b075:	48 89 fb             	mov    %rdi,%rbx
   2b078:	48 85 c0             	test   %rax,%rax
   2b07b:	74 25                	je     2b0a2 <util_buf_alloc+0x32>
   2b07d:	48 3b 43 18          	cmp    0x18(%rbx),%rax
   2b081:	74 0d                	je     2b090 <util_buf_alloc+0x20>
   2b083:	48 8b 10             	mov    (%rax),%rdx
   2b086:	48 89 53 10          	mov    %rdx,0x10(%rbx)
   2b08a:	5b                   	pop    %rbx
   2b08b:	c3                   	retq   
   2b08c:	0f 1f 40 00          	nopl   0x0(%rax)
   2b090:	48 c7 43 18 00 00 00 	movq   $0x0,0x18(%rbx)
   2b097:	00 
   2b098:	48 c7 43 10 00 00 00 	movq   $0x0,0x10(%rbx)
   2b09f:	00 
   2b0a0:	5b                   	pop    %rbx
   2b0a1:	c3                   	retq   
   2b0a2:	e8 29 fd ff ff       	callq  2add0 <util_buf_grow>
   2b0a7:	85 c0                	test   %eax,%eax
   2b0a9:	75 07                	jne    2b0b2 <util_buf_alloc+0x42>
   2b0ab:	48 8b 43 10          	mov    0x10(%rbx),%rax
   2b0af:	90                   	nop
   2b0b0:	eb cb                	jmp    2b07d <util_buf_alloc+0xd>
   2b0b2:	31 c0                	xor    %eax,%eax
   2b0b4:	5b                   	pop    %rbx
   2b0b5:	c3                   	retq   
   2b0b6:	66 2e 0f 1f 84 00 00 	nopw   %cs:0x0(%rax,%rax,1)
   2b0bd:	00 00 00
```
- `util_buf_alloc` with dlist:
```
000000000002b060 <util_buf_alloc>:
   2b060:	53                   	push   %rbx
   2b061:	48 8b 47 10          	mov    0x10(%rdi),%rax
   2b065:	48 8d 57 10          	lea    0x10(%rdi),%rdx
   2b069:	48 89 fb             	mov    %rdi,%rbx
   2b06c:	48 39 d0             	cmp    %rdx,%rax
   2b06f:	74 13                	je     2b084 <util_buf_alloc+0x24>
   2b071:	48 8b 08             	mov    (%rax),%rcx
   2b074:	48 8b 50 08          	mov    0x8(%rax),%rdx
   2b078:	48 89 0a             	mov    %rcx,(%rdx)
   2b07b:	48 8b 08             	mov    (%rax),%rcx
   2b07e:	48 89 51 08          	mov    %rdx,0x8(%rcx)
   2b082:	5b                   	pop    %rbx
   2b083:	c3                   	retq   
   2b084:	e8 47 fd ff ff       	callq  2add0 <util_buf_grow>
   2b089:	85 c0                	test   %eax,%eax
   2b08b:	75 06                	jne    2b093 <util_buf_alloc+0x33>
   2b08d:	48 8b 43 10          	mov    0x10(%rbx),%rax
   2b091:	eb de                	jmp    2b071 <util_buf_alloc+0x11>
   2b093:	31 c0                	xor    %eax,%eax
   2b095:	5b                   	pop    %rbx
   2b096:	c3                   	retq   
   2b097:	66 0f 1f 84 00 00 00 	nopw   0x0(%rax,%rax,1)
   2b09e:	00 00
```